cmd/zdb/zdb.c

```diff
@@ -6344,6 +6350,17 @@ dump_block_stats(spa_t *spa)
 		    100.0 * alloc / space);
 	}
 
+	if (spa_embedded_log_class(spa)->mc_allocator[0].mca_rotor != NULL) {
```

Comment:
Is the selection of the embedded_log_class dynamic? (i.e. at each spa_import?). I'm trying to understand how `zdb`  instance of the pool knows which ms to use for embedded log and if it can be different from what the runtime pool is using?

If `zdb` can know the actual ms, then a follow on zdb change could augment the `zdb -m` output by tagging the embedded log ms.

Reply:
Yes, the metaslab is selected when opening the pool.  So zdb wouldn't necessarily select the same metaslab as was used recently (or is currently in use, if the pool is imported while running zdb).  But typically, on pools that have moderate or higher fullness or fragmentation, the zil metaslab will really stick out as it will be the only one that's (nearly) empty and unfragmented.